### PR TITLE
Fix for issue #1685

### PR DESCRIPTION
**Pull request by bomber8013**
Tox tests pass. Fix seems to be working.

### CODE REVIEW

**Comment by aragilar** [comment]
@bomber8013 Can you add a test for this?

File: .authors.yml [hunk 1]
```diff
@@ -667,3 +667,10 @@
   alternate_emails:
     - alemigo@github.com
     - 32462069+alemigo@users.noreply.github.com
+- name: Alex Porter
+  email: bomber8013@gmail.com
+  github: bomer8013
```
Comment (takluyver) [inline .authors.yml:672]:
```suggestion
  github: bomber8013
```

:wink:

**Comment by takluyver** [comment]
Thanks for this. Have you checked if an expanded virtual dataset actually exposes new data? I think I recall that using a selection `source[:]` is equivalent to `source[0:len(source_dataset)]`, so you can grow the virtual dataset but the new space will be empty.

**Comment by bomber8013** [comment]
> Thanks for this. Have you checked if an expanded virtual dataset actually exposes new data? I think I recall that using a selection `source[:]` is equivalent to `source[0:len(source_dataset)]`, so you can grow the virtual dataset but the new space will be empty.

In writing the test for resizing virtual datasets, suggested by @aragilar , I find that what you describe does not happen. I am going to limit the scope of the test to just allowing virtual datasets to resize. I'll make another issue for the behavior or the resizing.

**Comment by bomber8013** [comment]
> > Thanks for this. Have you checked if an expanded virtual dataset actually exposes new data? I think I recall that using a selection `source[:]` is equivalent to `source[0:len(source_dataset)]`, so you can grow the virtual dataset but the new space will be empty.
> 
> In writing the test for resizing virtual datasets, suggested by @aragilar , I find that what you describe does not happen. I am going to limit the scope of the test to just allowing virtual datasets to resize. I'll make another issue for the behavior or the resizing.

In doing more investigation. Applying H5Dset_extent is actually something like a no-op. @takluyver the behavior you describe is taken care of in a snippet like this (adapted from [RFC:	HDF5 Virtual Dataset](https://support.hdfgroup.org/HDF5/docNewFeatures/VDS/HDF5-VDS-requirements-use-cases-2014-12-10.pdf)) examples:
```
with h5py.File("resize", "w") as f:
    source_dset = f.create_dataset(
        "source",
        data=np.arange(20),
        shape=(10, 2),
        maxshape=(None, 2),
        chunks=(10, 1),
        fillvalue=-1
    )
    layout = h5py.VirtualLayout((10, 1), np.int, (None, 1))
    layout_source = h5py.VirtualSource(source_dset)
    layout[:, 0] = layout_source[:, 1]
    for vspace, fpath, dset, src_space in layout.sources:
        vspace.select_hyperslab((0, 0), (h5py.h5s.UNLIMITED, 1))
        src_space.select_hyperslab((0, 1), (h5py.h5s.UNLIMITED, 1))
    
    virtual_dset: h5py.Dataset = f.create_virtual_dataset("virtual", layout)

print(source_dset[:, :])
print(virtual_dset[:, :])
```
Output:
```
[[ 0  1]
 [ 2  3]
 [ 4  5]
 [ 6  7]
 [ 8  9]
 [10 11]
 [12 13]
 [14 15]
 [16 17]
 [18 19]]
[[ 1]
 [ 3]
 [ 5]
 [ 7]
 [ 9]
 [11]
 [13]
 [15]
 [17]
 [19]]
```
Followed by:
```
source_dset.resize(20, 0)
print(source_dset[:, :])
print(virtual_dset[:, :])
```
Output:
```
[[ 0  1]
 [ 2  3]
 [ 4  5]
 [ 6  7]
 [ 8  9]
 [10 11]
 [12 13]
 [14 15]
 [16 17]
 [18 19]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]
 [-1 -1]]
[[ 1]
 [ 3]
 [ 5]
 [ 7]
 [ 9]
 [11]
 [13]
 [15]
 [17]
 [19]
 [-1]
 [-1]
 [-1]
 [-1]
 [-1]
 [-1]
 [-1]
 [-1]
 [-1]
 [-1]]
```
I would conclude that the changes I have made so far are pretty much useless, and even the original issue probably be closed and replaced by a new issue regarding the behavior of the virtual datasets. Let me know your thoughts.

**Comment by takluyver** [comment]
So IIUC, it's only possible to set up the 'UNLIMITED' selection you need for this to make sense using the low-level API (`.select_hyperslab()`), not with the high level API. That's not entirely useless, but it may cause some confusion.

My take is that virtual datasets are cheap to create, so if you add more source data, you may as well create a new virtual dataset and discard the old one. So support for unlimited selections & resizing isn't particularly interesting for me, but I'm not against adding that support if we can find a clean way to do it. Ideally that would mean both that it's clear for users and that it doesn't need much extra complexity internally.

See also #1202.

**Comment by bomber8013** [comment]
Thanks for showing me #1202 . After reviewing that, it would seem what I'm trying to get done falls under that ticket. With that in mind, I'm going to close this pull request and the associated issue and work on fixing that one. Thanks!

**Pull request closed without merging**